grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/plan/GraqlTraversalPlanner.java

```diff
@@ -105,14 +108,19 @@ private static Atom optimalCandidate(List<Atom> candidates){
             List<Atom> atomsToPlan = new ArrayList<>(atoms);
             atomsToPlan.remove(first);
```

Comment:
do we want to remove a possible `null` from a list?

Reply:
It shouldn't have nulls. `first` can be null only if `candidates` are empty. Basically we can move the if statement two lines up.